data/core/view.lua

```diff
@@ -91,57 +102,76 @@ function View:move_towards(t, k, dest, rate, name)
 end
 
 
+--- Called when view is requested to close (e.g., tab close button).
+--- Override to show confirmation dialogs for unsaved changes.
+--- @param do_close function Call this function to actually close the view
+--- Example: `core.command_view:enter("Save?", {submit = do_close})`
 function View:try_close(do_close)
   do_close()
 end
 
 
+--- Get the name displayed in the view's tab.
+--- Override to show document name, file path, etc.
 ---@return string
 function View:get_name()
   return "---"
 end
 
 
----@return number
+--- Get the total scrollable height of the view's content.
+--- Used by scrollbar to calculate thumb size and position.
+---@return number Height in pixels (default: infinite)
 function View:get_scrollable_size()
   return math.huge
 end
 
----@return number
+--- Get the total scrollable width of the view's content.
+--- Used by horizontal scrollbar.
+---@return number Width in pixels (default: 0, no horizontal scroll)
 function View:get_h_scrollable_size()
   return 0
 end
 
 
+--- Whether this view accepts text input (enables IME).
+--- Override and return true for text editors and input fields.
 function View:supports_text_input()
   return false
 end
 
----@param x number
----@param y number
----@return boolean
+--- Check if a screen point overlaps either scrollbar.
+--- Useful for determining cursor style or handling clicks.
+---@param x number Screen x coordinate
+---@param y number Screen y coordinate
+---@return boolean True if point is over vertical or horizontal scrollbar
 function View:scrollbar_overlaps_point(x, y)
   return not (not (self.v_scrollbar:overlaps(x, y) or self.h_scrollbar:overlaps(x, y)))
 end
 
 
----@return boolean
+--- Check if user is currently dragging either scrollbar.
+---@return boolean True if scrollbar drag is in progress
 function View:scrollbar_dragging()
   return self.v_scrollbar.dragging or self.h_scrollbar.dragging
 end
 
 
----@return boolean
+--- Check if mouse is hovering over either scrollbar track.
+---@return boolean True if mouse is over scrollbar
 function View:scrollbar_hovering()
   return self.v_scrollbar.hovering.track or self.h_scrollbar.hovering.track
 end
 
 
----@param button core.view.mousebutton
----@param x number
----@param y number
----@param clicks integer
----return boolean
+--- Handle mouse button press events.
+--- Override to handle clicks. Return true to consume event and prevent propagation.
+--- Base implementation handles scrollbar clicks.
+---@param button core.view.mousebutton "left" or "right"
+---@param x number Screen x coordinate
+---@param y number Screen y coordinate
+---@param clicks integer Number of clicks (1=single, 2=double, 3=triple)
```

Comment:
The maximum amount of consecutive clicks is actually configurable with `config.max_clicks` 

Maybe should be documented as `Number of consecutive clicks`